logs_sdk/lib/opentelemetry/sdk/logs/read_write_log_record.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Logs
+      # A {LogRecord} interface that can write to the full {LogRecord} and
+      # retrieve all information added to the {LogRecord}.
+      class ReadWriteLogRecord < OpenTelemetry::Logs::LogRecord
+        attr_accessor :timestamp,
+                      :observed_timestamp,
+                      :span_context,
+                      :severity_text,
+                      :severity_number,
+                      :body,
+                      :resource,
+                      :instrumentation_scope,
+                      :attributes
+
+        # Creates a new {ReadWriteLogRecord}.
+        #
+        # @param [optional Float, Time] timestamp Time when the event occurred.
+        # @param [optional Float, Time] observed_timestamp Time when the event
+        #   was observed by the collection system. If nil, will first attempt
+        #   to set to `timestamp`. If `timestamp` is nil, will set to
+        #   `Process.clock_gettime(Process::CLOCK_REALTIME)`.
+        # @param [optional OpenTelemetry::Trace::SpanContext] span_context The
+        #   OpenTelemetry::Trace::SpanContext to associate with the
+        #   {ReadWriteLogRecord}.
+        # @param [optional String] severity_text The log severity, also known as
+        #   log level.
+        # @param [optional Integer] severity_number The numerical value of the
+        #   log severity. See OpenTelemetry::Logs::SeverityNumber.
+        # @param [optional String, Numeric, Boolean, Array<String, Numeric,
+        #   Boolean>, Hash{String => String, Numeric, Boolean, Array<String,
+        #   Numeric, Boolean>}] body The body of the {ReadWriteLogRecord}.
+        # @param [optional Hash{String => String, Numeric, Boolean,
+        #   Array<String, Numeric, Boolean>}] attributes Attributes to associate
+        #   with the {ReadWriteLogRecord}.
+        # @param [OpenTelemetry::SDK::Logs::Logger] logger The logger that
+        #   created the {ReadWriteLogRecord}. Used to set `resource` and
+        #   `instrumentation_scope`.
+        #
+        # @return [ReadWriteLogRecord]
+        def initialize(
+          timestamp: nil,
+          observed_timestamp: nil,
+          span_context: nil,
+          severity_text: nil,
+          severity_number: nil,
+          body: nil,
+          attributes: nil,
+          logger: nil
+        )
+          @timestamp = timestamp
+          @observed_timestamp = observed_timestamp || timestamp || Process.clock_gettime(Process::CLOCK_REALTIME)
+          @span_context = span_context
```

Comment:
[The Logs Data Model](https://opentelemetry.io/docs/specs/otel/logs/data-model/#trace-context-fields) lists`trace_id`, `span_id`, and `trace_flags` as separate attributes. These attributes are all present in an `OpenTelemetry::SDK::Trace::SpanContext` object.

[The Logs SDK spec](https://opentelemetry.io/docs/specs/otel/logs/sdk/#readablelogrecord) says: 
> The [trace context fields](https://opentelemetry.io/docs/specs/otel/logs/data-model/#trace-context-fields) MUST be populated from the resolved Context (either the explicitly passed Context or the current Context) when [emitted](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#emit-a-logrecord)."

Each language treats this a little differently: 
- [Java: spanContext](https://github.com/open-telemetry/opentelemetry-java/blob/90ab3665dc0fb56598fcf9b9c15941eedb3822f8/sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkReadWriteLogRecord.java#L55)
- [Python: individual attrs](https://github.com/open-telemetry/opentelemetry-python/blob/d8490c5f557dd7005badeb800095cb51b553c98c/opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/__init__.py#L162-L164 )
- [PHP: context, spanContext](https://github.com/open-telemetry/opentelemetry-php/blob/62a14d1c20db456baabbabf003f5efdbf0b030da/src/SDK/Logs/ReadableLogRecord.php#L38-L39)
- [JS: SpanContext](https://github.com/open-telemetry/opentelemetry-js/blob/b400c2e5d9729c3528482781a93393602dc6dc9f/experimental/packages/sdk-logs/src/LogRecord.ts#L35)

What makes the most sense for Ruby? I'm leaning toward `SpanContext` and adding the current span context as the default. Is there a way to get the current span context? I've found a way to get the current `Context` via `Context.current`, but `Context` doesn't seem quite right since it doesn't necessarily have `trace_id`, `span_id`, and `trace_flags` as far as I can tell.